File: lista-de-exercicios/exercicio-3/Maquina.ts
import { Escolhas } from './Jokenpo.js';
import { JogadorBase } from './Jokenpo.js';

export class Maquina implements JogadorBase {
    private escolha: Escolhas;

    public getEscolha(): Escolhas {
        const aleatoria = Math.floor(Math.random() * 3);
        this.escolha = aleatoria;
        return aleatoria;
    }
}

File: lista-de-exercicios-java/src/main/java/org/example/exercicio03/Maquina.java
package org.example.exercicio03;

public class Maquina implements JogadorBase {
    private Escolha escolha;

    @Override
    public Escolha getEscolha() {
        int aleatoria = (int) Math.floor(Math.random() * 3);
        this.escolha = Escolha.values()[aleatoria];
        return escolha;
    }
}
